### PR TITLE
[`TF`] Fix Tensorflow XLA Generation on limited seq_len models

### PR DESCRIPTION
**Pull request by vasqu**
# What does this PR do?
Fixes xla generation with models with a limited seq_len that they can generate. It has been recently (re)discovered in #33298, which results in slow runs to fail as we have indexing issues based on absolute positional embeddings (i.e., an index out-of-bounds error). The issue can only be observed on CPUs and not on GPUs (idk why but the error is just silently swallowed). 

The solution is to extend the embedding not only by `max_new_tokens` but also by the current `seq_len` at hand. Atm, `seq_len` hasn't been considered yet.

No fixes but a ref to #33298
(possibly adding an issue if wanted)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@Rocketknight1 @amyeroberts

### CODE REVIEW

File: tests/test_modeling_tf_common.py [hunk 1]
```diff
@@ -1753,12 +1748,20 @@ def _generate_and_check_results(model, inputs_dict):
...
 
+            # extract the input to the model
+            is_input_ids = "input_ids" in inputs_dict
+            is_input_features = "input_features" in inputs_dict
+            if not (is_input_ids or is_input_features):
+                raise ValueError("No valid generate input found in inputs_dict")
+            inputs = inputs_dict["input_ids"] if is_input_ids else inputs_dict["input_features"]
+
             # fix config for models with additional sequence-length limiting settings
+            seq_len = inputs.get_shape()[1]
```
Comment (vasqu) [inline tests/test_modeling_tf_common.py:1759]: Not sure if input_features also has its sequence length at 1. Just submitting this quickly before I don't have time.

**Comment by Rocketknight1** [comment]
@vasqu I suspect the reason that these errors are silent on GPU is that `tf.gather()` [does not do bounds checking on GPU](https://www.tensorflow.org/api_docs/python/tf/gather):

> Caution: On CPU, if an out of bound index is found, an error is raised. On GPU, if an out of bound index is found, a 0 is stored in the corresponding output value. 

`tf.gather()` is the underlying mechanism used for a lot of things like embedding layers and other index lookups, so this affects a lot of layers.

Either way, reviewing the rest of this now!

**Comment by Rocketknight1** [comment]
@vasqu can you push another empty commit with the `[run-slow] opt` message to trigger the slow CI there?

**Comment by vasqu** [comment]
Oh wow, thanks for the insights into this. This is rather counter-intuitive to me tho :D Even some archaic cuda / gpu errors would suffice for me but it is what it is.

**Comment by vasqu** [comment]
tests pass :) gentle ping @Rocketknight1

**Review by Rocketknight1 — APPROVED** [review]
LGTM!

For core maintainers: This generation test fails on some models because it generates past the limit of the model's `max_position_embeddings`. Because the model is initialized from a config, not a pretrained checkpoint, this PR simply increases `config.max_position_embeddings` or `config.max_target_positions`, which seems like the right fix to me.

cc @lysandrejik for core maintainer review!

**Review by ArthurZucker — APPROVED** [review]
Thanks 🤗

**Pull request merged**